File: env_loader.php
<?php
/**
 * Cargador de variables de entorno desde archivo .env
 * Versión simple sin dependencias externas
 */

class EnvLoader {
    private static $loaded = false;
    private static $env = [];
    
    public static function load($path = null) {
        if (self::$loaded) {
            return;
        }
        
        if ($path === null) {
            $path = __DIR__ . '/.env';
        }
        
        if (!file_exists($path)) {
            throw new Exception("Archivo .env no encontrado en: " . $path);
        }
        
        $lines = file($path, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
        
        foreach ($lines as $line) {
            // Ignorar comentarios
            if (strpos($line, '#') === 0) {
                continue;
            }
            
            // Buscar líneas con formato KEY=VALUE
            if (strpos($line, '=') !== false) {
                list($key, $value) = explode('=', $line, 2);
                
                $key = trim($key);
                $value = trim($value);
                
                // Remover comillas si existen
                if ((substr($value, 0, 1) === '"' && substr($value, -1) === '"') ||
                    (substr($value, 0, 1) === "'" && substr($value, -1) === "'")) {
                    $value = substr($value, 1, -1);
                }
                
                // Guardar en array interno y en $_ENV
                self::$env[$key] = $value;
                $_ENV[$key] = $value;
                putenv("$key=$value");
            }
        }
        
        self::$loaded = true;
    }
    
    public static function get($key, $default = null) {
        self::load();
        return isset(self::$env[$key]) ? self::$env[$key] : $default;
    }
    
    public static function getAll() {
        self::load();
        return self::$env;
    }
}

/**
 * Función helper para obtener variables de entorno
 */
function env($key, $default = null) {
    return EnvLoader::get($key, $default);
}
?>

File: scripts/python/env_loader.py
"""
Cargador de variables de entorno para Python
Equivalente al env_loader.php
"""
import os
from typing import Any, Optional, Dict

class EnvLoader:
    _loaded = False
    _env = {}
    
    @classmethod
    def load(cls, path: Optional[str] = None) -> None:
        if cls._loaded:
            return
        
        if path is None:
            path = os.path.join(os.path.dirname(__file__), '.env')
        
        if not os.path.exists(path):
            raise FileNotFoundError(f"Archivo .env no encontrado en: {path}")
        
        with open(path, 'r', encoding='utf-8') as file:
            for line in file:
                line = line.strip()
                
                # Ignorar líneas vacías y comentarios
                if not line or line.startswith('#'):
                    continue
                
                # Buscar líneas con formato KEY=VALUE
                if '=' in line:
                    key, value = line.split('=', 1)
                    key = key.strip()
                    value = value.strip()
                    
                    # Remover comillas si existen
                    if ((value.startswith('"') and value.endswith('"')) or
                        (value.startswith("'") and value.endswith("'"))):
                        value = value[1:-1]
                    
                    # Guardar en diccionario interno y en variables de entorno
                    cls._env[key] = value
                    os.environ[key] = value
        
        cls._loaded = True
    
    @classmethod
    def get(cls, key: str, default: Any = None) -> Any:
        cls.load()
        return cls._env.get(key, default)
    
    @classmethod
    def get_all(cls) -> Dict[str, str]:
        cls.load()
        return cls._env.copy()

def env(key: str, default: Any = None) -> Any:
    """Función helper para obtener variables de entorno"""
    return EnvLoader.get(key, default)

def get_db_config():
    """Obtener configuración de base de datos desde .env"""
    return {
        'host': env('DB_HOST', 'localhost'),
        'database': env('DB_NAME', 'chatBeto'),
        'user': env('DB_USERNAME', 'root'),
        'password': env('DB_PASSWORD', ''),
        'charset': 'utf8mb4'
    }

def get_app_config():
    """Obtener configuración de aplicación desde .env"""
    return {
        'name': env('APP_NAME', 'ChatBETO'),
        'environment': env('APP_ENV', 'production'),
        'debug': env('APP_DEBUG', 'false').lower() in ('true', '1', 'yes')
    }